test/data/javascript.js

```diff
@@ -104,6 +107,218 @@ exports.test_data = {
 
         ],
     }, {
+        name: "operator_position option - ensure no neswlines if preserve_newlines is false",
+        matrix: [
+            {
+                options: [
+                    { name: "operator_position", value: "'before_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'after_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }, {
+                options: [
+                    { name: "operator_position", value: "'preserve_newline'" },
+                    { name: "preserve_newlines", value: "false" }
+                ]
+            }
+        ],
+        tests: [
+            {
+                unchanged: inputlib.operator_position.sanity
+            }, {
+                input: inputlib.operator_position.comprehensive,
+                output: inputlib.operator_position.sanity,
+            }
+        ]
+    }, {
+        name: "reset preserve_newlines and operator_position",
+        options: [
+            { name: "preserve_newlines", value: "true" },
+            { name: "operator_position", value: "'before_newline'" }
+        ]
+    }, {
+        name: "operator_position option - set to 'before_newline' (default value)",
+        tests: [
+            {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive,
+                output: [
+                    'var res = a + b - c /',
+                    '    d * e %',
+                    '    f;',
```

Comment:
So I spent some time on this tonight and actually don't agree anymore.  The reason is that if the following holds true:
```
// operator_position = before_newline
a
|| b

// resolves to

a ||
  b
```
Then people will expect the following:
```
// operator_position = after_newline

a ||
b

// resolves to

a
  || b
```

With my understanding of how js-beautify parses text, this last resolution would only be possible by waiting to process the 'b' token.  We'd use the 'wanted_newline' flag to then add a newline before the previous operator token.  This seems messy and unnecessary when preserving the operator_position seems intuitive enough.

Thoughts?